test/e2e/e2e.go

```diff
@@ -2376,31 +2376,34 @@ var _ = ginkgo.Describe("host to host-networked pods traffic validation", func()
 							continue
 						}
 
-						responses := sets.NewString()
-						// Fill expected responses, it should hit the nodeLocal endpoints only and not SNAT packet IP
-						expectedResponses := sets.NewString()
-
-						expectedResponses.Insert(node.Name, nodeAddress.Address)
-
 						valid := false
 						nodePort := nodeTCPPort
+						var epHostname string
+						var ip string
 
-						ginkgo.By("Hitting the nodeport on " + node.Name + " and trying to reach only the local endpoint with protocol " + protocol)
+						ginkgo.By(fmt.Sprintf("Hitting nodeport %d on %s and trying to reach only the local endpoint with protocol %s", nodePort, node.Name, protocol))
 						for i := 0; i < maxTries; i++ {
-							epHostname := curlInContainer(node.Name, protocol, nodeAddress.Address, nodePort, "hostname")
-							epClientIP := curlInContainer(node.Name, protocol, nodeAddress.Address, nodePort, "clientip")
-							ip, _, err := net.SplitHostPort(epClientIP)
+							epHostname, err = curlInContainer(node.Name, protocol, nodeAddress.Address, nodePort, "hostname", 60)
```

Comment:
Actually, if this times out, it will fail without looking at the retry (see the line before). That's how it worked before, and I kept that part of the logic. 
The retry is only used if `epHostname == node.Name && ip == nodeAddress.Address` evaluates to false. I'll lower the timeout to 10 seconds, just in case (because this is not a connect-timeout, it's the timeout for the entire session (connection establishment + the transfer).